0003/README.md

```diff
@@ -0,0 +1,123 @@
+- For this RFD log-server address can be a folder in the provisioner. If the user wants he can mount to a separate docker container.
+- We would like this framework to have a way where a user change the logging configuration like log-server address or log-driver. For example let say, if the default log-driver is `syslog` then the user can configure it to`fluentbit`, add some other services like ELK and kibana to extend the framework. 
+
+## Progress:
+
+This implementation of framework has been tested with rsyslog with a POC and been discussed in-depth within the Packet and Infracloud team.
+
+## Suggestions:
+
+- By default, the logging feature can be supported by rsyslog which is built on the top of syslog.
+- We want to ensure at a basic minimum there is a centralized logging framework which exists for Tinkerbell. In order to have a generic implementation of configuring log-drivers below are some proposals.
+-) How to make action containers run with logging drivers.
+a) Place the logging details inside the workflow template. 
+```
+	version: "0.1"
+	name: hello_world_workflow
+	global_timeout: 600
+	tasks:
+	  - name: "hello world"
+	    worker: "{{.device_1}}"
+	    actions:
+	      - name: "hello_world"
+		image: hello-world
+		timeout: 60  
+	logging:
+	  type: syslog  # log driver
+	  config:
+	    syslog-address : tcp://192.168.1.1:514
+	    tag: "container_name/{{.Name}}"
+ ```
+ 
+ - This will ensure all the worflow action containers have consistent way of logging informations. 
+ - logging configuration would be specific to workflow containers. 
+
+
+b) Host level configuration. 
+- We can have a file for host level configuration. This will set environment variables which can be used in action.go while creating the containes. 
+- This will have a way where all the workflow template will have the common logging framework.
+- Problem
+    - Design issue: logging here is specific to action docker containers. So having them in the workflow template makes more sense other than in separate config file. 
+    - while maintaining multi-node cluster, logging configuration can vary.
+    - updating the logging configuration. 
+
+
+c) Workaround 
+   - Have configuration defined in action.go when the action container is created. This will ensure we have the bare centralized framework ready.
+   - Challenge:  
+     - change in implementation when changes are done in provisioner. Like, moving the whole infrastructure on a private network. 
+
+## System Context Diagram
+
+Example with rsyslog
+
+![rsyslog](rsyslog.png)
+
+## Future roadmap:
+
+As a part of roadmap implementation, certain changes in the existing framework would be required to bring below proposals onboard.
+
+1) Implementation of Distribute tracing on the microservices. 
```

Comment:
Let's leave tracing for another proposal

Reply:
I have added it in the road map. Should we remove it from there?

Reply:
I think we should not mention tracing at all here

Reply:
i have removed it for now. We can take it up as separate RFD.